blue_ai/envs/transient_goals.py

```diff
@@ -52,88 +57,114 @@ def __init__(
         self.transient_penalty = transient_penalty
         self.n_transient_obstacles = n_transient_obstacles
         self.transient_obstacles = transient_obstacles
-        self.replace_transient_obstacles = replace_transient_goals
+        self.replace_transient_obstacles = replace_transient_obstacles
 
         mission_space = MissionSpace(mission_func=self._gen_mission)
 
-        max_steps = 4 * width**2
+        ## This seems really arbitrary
+        max_steps = 4 * (len(self.im[0])) ** 2
 
         super().__init__(
             mission_space=mission_space,
-            width=width,
-            height=height,
+            width=len(self.im[0]),
+            height=len(self.im),
             # Set this to True for maximum speed
-            see_through_walls=True,
+            see_through_walls=False,
```

Comment:
If we want to mess with the environments more, such as adding more walls within the env, it's worth letting this small performance gain go.